pkg/build/api/types.go

```diff
@@ -966,6 +987,9 @@ type BuildRequest struct {
 	// TriggeredBy describes which triggers started the most recent update to the
 	// buildconfig and contains information about those triggers.
 	TriggeredBy []BuildTriggerCause
+
+	// StrategyOptions contains additional strategy-specific options for the build
+	StrategyOptions *BuildStrategyOptions
```

Comment:
how would the strategies implement that interface?  you'd have to introduce getters and it would have to be the same getters on every implementation, implying they have the same fields.

we can introduce wrapper structs if desired in the future for passing things between methods, they don't have to be part of the api though.